File: abstractions/typescript/src/getPathParameters.ts
import { RequestInformation } from "./requestInformation";

export function getPathParameters(parameters: Record<string, unknown> | string | undefined) : Record<string, unknown> {
    const result:  Record<string, unknown> = {};
    if(typeof parameters === "string") {
        result[RequestInformation.raw_url_key] = parameters;
    } else if(parameters) {
        for(const key in parameters){
            result[key]= parameters[key];
        };
    }
    return result;
}

File: abstractions/python/kiota/abstractions/get_path_parameters.py
from typing import Any, Dict, Optional, Union

from .request_information import RequestInformation


def get_path_parameters(parameters: Union[Dict[str, Any], Optional[str]]) -> Dict[str, Any]:
    result: Dict[str, Any] = {}
    if isinstance(parameters, str):
        result[RequestInformation.RAW_URL_KEY] = parameters
    elif isinstance(parameters, dict):
        for key, val in parameters.items():
            result[key] = val
    return result
